### PR TITLE
[dazai] Weight-decay sweep {3e-4, 1e-3, 3e-3} on alphonse Fourier base

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The bengio Wave 1 baseline `m9775k1v` (alphonse 4L/256d + Fourier PE + no-EMA + T_max=30) shows clear evidence of overfitting: val_volume_pressure=4.17% beats the AB-UPT target of 6.08%, but test_volume_pressure on best-checkpoint evals comes in at ~12.9% (~3x the val number). Surface metrics show a similar but smaller val/test gap (e.g., test sp ~5.1% vs val sp ~4.8%). The model is memorizing case-specific volumetric patterns from the 400-case train set rather than learning a generalizing surrogate. This overfitting signal is the cleanest, most-actionable lever we have not yet exploited on the bengio branch.

The current recipe uses `weight_decay=1e-4` (default). frieren PR #310 was assigned this hypothesis but never reported a single result before being closed for non-response, so the wd-sweep is still **completely untested** on bengio. We hypothesize that **stronger weight decay attacks the val/test gap directly** — particularly on volume pressure — by penalizing the model's ability to memorize per-case volumetric features in the FFN/attention weights. The cost is a small expected hit to val abupt; the prize is a substantially better test abupt and a closer val/test gap.

This is a **plumbing-only** experiment: pure CLI flag sweep on the Wave 1 base. No code edits. Single hypothesis, single axis, three trials.

## Instructions

Run a sequential 3-trial weight-decay sweep on the alphonse Wave 1 4L/256d + Fourier PE recipe. **All other flags must match the baseline exactly** — the only thing that changes between trials is `--weight-decay`. Use a single sequential orchestrator (not parallel) so each trial gets the full 4 GPUs.

**Trials (sweep `--weight-decay`):**
- **Trial A**: `--weight-decay 3e-4` (3x baseline)
- **Trial B**: `--weight-decay 1e-3` (10x baseline — frieren's intended target value)
- **Trial C**: `--weight-decay 3e-3` (30x baseline — aggressive)

**Reproduce command (Trial A — change `--weight-decay` for B and C):**

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --weight-decay 3e-4 \
  --lr-cosine-t-max 30 \
  --epochs 30 \
  --no-compile-model \
  --kill-thresholds "89000:val_primary/abupt_axis_mean_rel_l2_pct<11.5,178000:val_primary/abupt_axis_mean_rel_l2_pct<9.0" \
  --wandb-group bengio-wave8-dazai-wd-sweep \
  --wandb-name "wd3e-4-trialA" \
  --agent dazai
```

For Trial B: `--weight-decay 1e-3 --wandb-name "wd1e-3-trialB"`.
For Trial C: `--weight-decay 3e-3 --wandb-name "wd3e-3-trialC"`.

**Sanity checks at launch:**
- `n_params == 3,249,813` (confirms FourierEmbed is wired correctly — the 12,544 param gap from 3,237,269 is the FourierEmbed `Linear(48→256)` projection)
- DDP world_size=4
- LR cosine schedule: 3e-4 → ~0 by ep30
- AdamW, no EMA, no dropout

**Kill-threshold operator semantics (CRITICAL — read carefully):**
The `<VALUE` operator means "metric must remain below VALUE; kill if not". This is the divergence/quality guard semantics. Do NOT use `>VALUE` (that inverts the gate). See `trainer_runtime.py:723-732` and PR #325/#328 for the bug history.

**Sequential orchestration:**
Run a bash loop:
```bash
for trial in A B C; do
  case $trial in
    A) WD=3e-4; NAME="wd3e-4-trialA";;
    B) WD=1e-3; NAME="wd1e-3-trialB";;
    C) WD=3e-3; NAME="wd3e-3-trialC";;
  esac
  torchrun --standalone --nproc-per-node=4 train.py \
    --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
    --fourier-pe --no-use-ema --lr 3e-4 --weight-decay $WD \
    --lr-cosine-t-max 30 --epochs 30 --no-compile-model \
    --kill-thresholds "89000:val_primary/abupt_axis_mean_rel_l2_pct<11.5,178000:val_primary/abupt_axis_mean_rel_l2_pct<9.0" \
    --wandb-group bengio-wave8-dazai-wd-sweep --wandb-name "$NAME" --agent dazai
done
```

**Reporting cadence:**
- Within 1 hour of launch: post Trial A W&B run ID + ep1 sanity (alphonse `m9775k1v` ep1 was 13.50%; expected range 13.0–14.5%).
- ep5 gate: post `abupt`, `surface_pressure`, `wall_shear`, `vol_p`, `wsx`, `wsy`, `wsz` for Trial A.
- ep10 gate: same. **If Trial A ep10 abupt > 9.0% (kill threshold), the run will be killed automatically — proceed to Trial B without manual intervention.**
- Trial completion: post the full per-channel val table at ep30 (best-checkpoint val), plus the W&B run ID, **for every trial**.
- Final summary: combined results table comparing all 3 trials side-by-side with the alphonse `m9775k1v` baseline (val_abupt=7.2091%, val_sp=4.802%, val_ws=8.160%, val_wsx=7.109%, val_wsy=9.100%, val_wsz=10.869%, val_vol_p=4.166%).

**Decision criteria (you do not need to wait for advisor input on these):**
- If any trial val_abupt at ep30 < 7.21%: **win** — that wd value beats baseline. Continue.
- If Trial C (wd=3e-3) val_abupt at ep30 > 8.5% but val/test_vol_p gap shrinks substantially: **partial win** — over-regularized for val but the regularization signal works. Recommend Trial D at wd=2e-3.
- If all 3 trials are worse on val_abupt AND show no improvement on the val/test_vol_p gap: hypothesis falsified — close PR with full negative-result table.

**Test-set evaluation (CRITICAL):**
Volume pressure has a known ~3x val/test gap on this architecture. **For the best-checkpoint val_abupt trial, also run `python eval.py` (or the test-eval harness in your pod) on the best checkpoint** and report the test_primary/* metrics in the final summary. Without test confirmation we cannot claim a true win on the val/test gap hypothesis.

## Baseline

Current best on bengio: **alphonse PR #74 `m9775k1v`** (4L/256d, FourierEmbed, no-EMA, lr=3e-4, T_max=30, weight_decay=1e-4 default).

| Metric | Val (`m9775k1v` ep30) | AB-UPT Target | Test (best-ckpt) |
|---|---:|---:|---:|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **7.2091** | 4.51 | ~8.48 |
| `val_primary/surface_pressure_rel_l2_pct` | 4.802 | 3.82 | ~5.08 |
| `val_primary/wall_shear_rel_l2_pct` | 8.160 | 7.29 | — |
| `val_primary/volume_pressure_rel_l2_pct` | **4.166** ✓ | 6.08 | **~12.90** ⚠ (3x val/test gap) |
| `val_primary/wall_shear_x_rel_l2_pct` | 7.109 | 5.35 | ~7.95 |
| `val_primary/wall_shear_y_rel_l2_pct` | 9.100 | 3.65 | ~10.90 |
| `val_primary/wall_shear_z_rel_l2_pct` | 10.869 | 3.63 | ~11.66 |

**The vol_p val/test gap is the primary target of this experiment. A successful trial reduces test_vol_p toward 6.08% while not hurting val_abupt by more than 0.2pp.**

W&B run: `m9775k1v` (group `alphonse-fourier-pe-baseline`).

ETA: ~28 min/epoch × 30 epochs × 3 trials ≈ 42 hours sequential. Comfortable within the 72h budget.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #348 because ADVISOR — Closing: no `senpai-bengio-dazai` pod exists in the cluster, so this PR will never be picked up by a student worker.

Confirmed via `kubectl get deployments -l app=senpai` — only the following bengio students are deployed: alphonse, askeladd, chihiro, edward, emma, fern, frieren, gilbert, haku, kohaku, nezuko, norman, senku, tanjiro, thorfinn, violet. There is no dazai deployment.

The weight-decay sweep hypothesis (3e-4 / 1e-3 / 3e-3 on alphonse Wave 1 base) remains scientifically valuable — it directly targets the val/test gap on volume_pressure (val 4.17% vs test ~12.9% on the m9775k1v baseline) and frieren PR #310 was closed for non-response without a single trial run. **This will be reassigned to a real pod in Wave 9.** Candidates: any student that becomes idle after the current wave concludes..
